fdbserver/KeyValueStoreRocksDB.actor.cpp

```diff
@@ -455,8 +455,12 @@ class ReadIteratorPool {
 	void update() {
 		if (SERVER_KNOBS->ROCKSDB_READ_RANGE_REUSE_ITERATORS ||
 		    SERVER_KNOBS->ROCKSDB_READ_RANGE_REUSE_BOUNDED_ITERATORS) {
-			std::lock_guard<std::mutex> lock(mutex);
-			iteratorsMap.clear();
+			mutex.lock();
+			// The latest index might contain the current iterator which is getting created.
+			// But, that should be ok to avoid adding more code complexity.
+			deletedUptoIndex = index;
+			mutex.unlock();
+			deleteIteratorsPromise.send(Void());
```

Comment:
In flow, the `send` here means we synchronously invoke the callback, i.e., `readIterPool->refreshIterators();` in the `refreshReadIteratorPool()`, so this doesn't seem to solve the problem.